From 99c614a21d07287874f853cc485e1acd053f4a47 Mon Sep 17 00:00:00 2001
From: Tianshu Qiu <tian.shu.qiu@intel.com>
Date: Thu, 24 May 2018 17:27:30 +0800
Subject: [PATCH 1951/2345] media: intel-ipu4: isys: Fix ISYS ABI regression

Message for Open Source:
crop_first_and_last_lines is not a valid field in
struct ipu_fw_isys_input_pin_info_abi. Remove it.

Message for Internal:
[Issue/Feature]
crop_first_and_last_lines is only valid for ipu6.
ipu4/ipu4p does not have field defined in ABI.
[Root Cause/Changes]
Remove crop_first_and_last_lines.

Change-Id: Ia4d77617c6b22f1d384710267e884f24b1e25f66
Signed-off-by: Tianshu Qiu <tian.shu.qiu@intel.com>
Signed-off-by: Meng Wei <wei.meng@intel.com>
---
 drivers/media/pci/intel/ipu-fw-isys.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/media/pci/intel/ipu-fw-isys.h b/drivers/media/pci/intel/ipu-fw-isys.h
index 2a10721..8ef68d4 100644
--- a/drivers/media/pci/intel/ipu-fw-isys.h
+++ b/drivers/media/pci/intel/ipu-fw-isys.h
@@ -563,10 +563,12 @@ struct ipu_fw_isys_param_pin_abi {
  *		     (enum ipu_fw_isys_mipi_store_mode)
  * @bits_per_pix: native bits per pixel
  * @mapped_dt: actual data type from sensor
+#if !defined(CONFIG_VIDEO_INTEL_IPU4) && !defined(CONFIG_VIDEO_INTEL_IPU4P)
  * @crop_first_and_last_lines    Control whether to crop the
  *                              first and last line of the
  *                              input image. Crop done by HW
  *                              device.
+#endif
  */
 struct ipu_fw_isys_input_pin_info_abi {
 	struct ipu_fw_isys_resolution_abi input_res;
@@ -574,7 +576,9 @@ struct ipu_fw_isys_input_pin_info_abi {
 	u8 mipi_store_mode;
 	u8 bits_per_pix;
 	u8 mapped_dt;
+#if !defined(CONFIG_VIDEO_INTEL_IPU4) && !defined(CONFIG_VIDEO_INTEL_IPU4P)
 	u8 crop_first_and_last_lines;
+#endif
 };
 
 /**
-- 
2.7.4

